### ISSUE TITLE
Environment variables don't seem to be available at build time

### ISSUE THREAD

**Issue by simoneb** [google-github-actions/deploy-cloudrun#460]
### TL;DR

I'm deploying from source a React app which needs a build-time environment variable. The build process is correctly executing my package.json `build` script, and the env var is provided as an input to the action, yet the env variable doesn't seem to be available when building the client side app but only at runtime, preventing me from being able to correctly build the client side app.

### Expected behavior

I would expect that provided environment variable are available at build time as well, rather than just at runtime

### Observed behavior

The environment variables are not available at build time

### Action YAML

```yaml
- id: 'deploy'
        uses: 'google-github-actions/deploy-cloudrun@v1'
        with:
          service: 'my-service'
          region: 'europe-west1'
          env_vars: |
            REACT_APP_GOOGLE_MAPS_API_KEY=${{ secrets.REACT_APP_GOOGLE_MAPS_API_KEY }}
          flags: --allow-unauthenticated
          source: .
```

### Log output

```text
There aren't any logs relevant to this issue
```

### Additional information

If this is known to be unsupported, is there a way to specify build time environment variables without having to set up Cloud Build manually?

**Comment by sethvargo** [comment]
Cloud Run does not currently support specifying build environment variables via `gcloud run deploy`: https://cloud.google.com/run/docs/deploying-source-code

So unfortunately you would need to build the container yourself. Note that you don't _have_ to youse Cloud Build - you can use any system capable of building a container, including GitHub Actions. You'll just need to push your container to Google Artifact Registry so that Cloud Run can use it to deploy. That would look something like:

```yaml
- id: 'build'
  run: npm build

- id: 'package'
  run: docker build ...

- id: 'publish'
  run: docker push ...
```

I'm omitting details like authentication, but that's the high-level architecture.

**Comment by simoneb** [comment]
> Cloud Run does not currently support specifying build environment variables via `gcloud run deploy`: https://cloud.google.com/run/docs/deploying-source-code
> 
> So unfortunately you would need to build the container yourself. Note that you don't _have_ to youse Cloud Build - you can use any system capable of building a container, including GitHub Actions. You'll just need to push your container to Google Artifact Registry so that Cloud Run can use it to deploy. That would look something like:
> 
> ```yaml
> - id: 'build'
>   run: npm build
> 
> - id: 'package'
>   run: docker build ...
> 
> - id: 'publish'
>   run: docker push ...
> ```
> 
> I'm omitting details like authentication, but that's the high-level architecture.

Thanks @sethvargo , I wasn't sure but I thought I'd ask anyway, and thanks for clarifying
